src/NuGet.Core/NuGet.Protocol/Resources/MetadataResourceV3.cs

```diff
@@ -25,18 +24,8 @@ public class MetadataResourceV3 : MetadataResource
         public MetadataResourceV3(HttpSource client, RegistrationResourceV3 regResource)
             : base()
         {
-            if (client == null)
-            {
-                throw new ArgumentNullException("client");
-            }
-
-            if (regResource == null)
-            {
-                throw new ArgumentNullException("regResource");
-            }
-
-            _regResource = regResource;
-            _client = client;
+            _regResource = regResource ?? throw new ArgumentNullException(nameof(regResource));
+            _client = client ?? throw new ArgumentNullException(nameof(client));
```

Comment:
the client is not used, so you can probably remove it. 
I know it's public, but I doubt anyone depends on this type explicitly.

Reply:
Done